Bottom: 47a1ff121d78e6106841040006b76dc86ea7b5f4
Top:    400e95a23e7148ec9a2269cd7d2866d99bdbd49e
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-07-27 00:34:12 +0100

hid/ps: Fix arc drawing since pcb_printf patch

The last parameter to the arc macro we use for drawing arcs is unitless,
so should be a double. This was broken by:

commit fa9ae1f6b1eabaab961795ce7be53afe46eaa735

    Convert ps/eps/lpr HID's to use pcb-printf

I've reverted the change, and added an extra (double) cast compared
to what was there before, I'm not sure we wouldn't get an integer
division without it. One day I should learn these type promotion /
casting rules properly! ;)


---

diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index f2e0706..201b721 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -1055,9 +1055,9 @@ ps_draw_arc (hidGC gc, int cx, int cy, int width, int height,
 	  cx, cy, width, height, start_angle, delta_angle, sa, ea);
 #endif
   use_gc (gc);
-  pcb_fprintf (global.f, "%d %d %mi %mi %mi %mi %mi a\n",
+  pcb_fprintf (global.f, "%d %d %mi %mi %mi %mi %g a\n",
                sa, ea, -width, height, cx, cy,
-               (global.linewidth + 2 * global.bloat) / width);
+               (double) (global.linewidth + 2 * global.bloat) / (double) width);
 }
 
 static void
